common/kv_db.hpp

```diff
@@ -12,16 +12,97 @@
 //    See the License for the specific language governing permissions and
 //    limitations under the License.
 
-#ifndef KV_DB_HPP
-#define KV_DB_HPP
+#ifndef MENDER_COMMON_KV_DB_HPP
+#define MENDER_COMMON_KV_DB_HPP
+
+#include <common/error.hpp>
+#include <common/expected.hpp>
+#include <config.h>
+
+#include <string>
+#include <mutex>
+#include <vector>
+#include <map>
+#include <functional>
+
+namespace mender::common::kv_db {
+
+using namespace std;
+
+namespace expected = mender::common::expected;
+
+enum KVDBErrorCode {
+	NoError = 0,
+	ParseError,
+	KeyError,
+};
+using KVDBError = mender::common::error::Error<KVDBErrorCode>;
+
+class DBEntry {
+public:
+	std::vector<std::byte> buf;
+	std::string key;
+
+public:
+	DBEntry() {
+	}
+	DBEntry(const string &key) :
+		key {key} {
+	}
+	DBEntry(const std::vector<std::byte> &val) :
+		buf {val} {
+	}
+	DBEntry(const string &key, const std::vector<std::byte> &val) :
+		key {key},
+		buf {val} {
+	}
+	void Write(const string &value) {
+		for (char c : value) {
+			buf.push_back(std::byte(c));
+		}
+	}
+
+	std::vector<std::byte> Read() {
+		return buf;
+	}
+};
+
+using ExpectedEntry = expected::Expected<DBEntry, KVDBError>;
+using ExpectedBool = expected::Expected<bool, KVDBError>;
+
+class KVDB {
```

Comment:
@oleorhagen: Another RFC: I think following [this principle](https://google.github.io/styleguide/cppguide.html#General_Naming_Rules) we should spell out this name rather than abbreviate it. Agree?

I like the name better than the original `Store` which was used in Golang though.

Reply:
Yes, agree completely!